### PR TITLE
Finishing Up Peer Transfers

### PR DESCRIPTION
**Pull request by colinthomas-z80**
Not finished yet, do not merge

### CODE REVIEW

**Comment by dthain** [comment]
Rebase on master, and your unpack woes should be gone now.

**Comment by dthain** [comment]
@colinthomas-z80 these changes make sense to me, please rebase on master.
Also, please use the `vine_tune` interface to set the number of transfers allowed.

**Comment by colinthomas-z80** [comment]
> @colinthomas-z80 what's the status on this one?

need to rebase and verify functionality

File: taskvine/src/manager/vine_manager.c [hunk 1]
```diff
@@ -2642,7 +2642,7 @@ static struct vine_task *vine_manager_sched_worker_transfer(struct vine_manager
 					if((remote_info = hash_table_lookup(peer->current_files, f->cached_name)) && remote_info->in_cache)
 					{
 						char *peer_source =  string_format("worker://%s:%d/%s", peer->transfer_addr, peer->transfer_port, f->cached_name);
-						if(vine_current_transfers_source_in_use(q, peer_source) < VINE_FILE_SOURCE_MAX_TRANSFERS)
+						if(vine_current_transfers_source_in_use(q, peer_source) < VINE_WORKER_SOURCE_MAX_TRANSFERS)
```
Comment (dthain) [inline taskvine/src/manager/vine_manager.c:2645]: Use file_source_max_transfers here?

File: taskvine/src/manager/vine_manager.c [hunk 2]
```diff
@@ -3236,11 +3237,18 @@ int vine_enable_monitoring_full(struct vine_manager *q, char *monitor_output_dir
...
 }
 
-int vine_enable_peer_transfers(struct vine_manager *q) {
+int vine_enable_peer_transfers(struct vine_manager *q) 
+{
 	q->peer_transfers_enabled = 1;
 	return 1;
 }
 
+int vine_set_file_source_max_transfers(struct vine_manager *q, int c)
```
Comment (dthain) [inline taskvine/src/manager/vine_manager.c:3246]: Roll this into the `vine_tune` interface, which covers most integer knobs.

File: taskvine/src/worker/vine_worker.c [hunk 3]
```diff
@@ -1084,7 +1084,7 @@ static int task_resources_fit_now( struct vine_task *t )
 	return
 		(cores_allocated  + t->resources_requested->cores  <= local_resources->cores.total) &&
 		(memory_allocated + t->resources_requested->memory <= local_resources->memory.total) &&
-		(disk_allocated   + t->resources_requested->disk   <= local_resources->disk.total) &&
+		(1) && //disk_allocated   + t->resources_requested->disk   <= local_resources->disk.total) &&
```
Comment (dthain) [inline taskvine/src/worker/vine_worker.c:1087]: If this is a temporary hack, remove it.  But if it is fixing a bug, add a big explanation here.

**Review by dthain — CHANGES_REQUESTED** [review]

File: taskvine/src/manager/vine_manager.c (hunk 1 above)
Comment (colinthomas-z80) [reply taskvine/src/manager/vine_manager.c:2645]: For now, workers still have only 1 transfer port open. So they may have only one connection.

**Review by dthain — APPROVED** [review]

**Pull request merged**